File: lab08/lab08/Models/Methods.cs
using System.Numerics;

namespace lab08.Models
{
    public class Methods
    {
        public async static Task<int[][]> MultiplyMatrix(int[][] matrix1, int[][] matrix2)
        {

            int[][] result = new int[matrix1.Length][];

            Parallel.For(0, matrix1.Length, i =>
            {
                result[i] = new int[matrix2[0].Length];
                for (int j = 0; j < matrix2[0].Length; j++)
                {
                    result[i][j] = 0;
                    for (int k = 0; k < matrix1[0].Length; k++)
                    {
                       result[i][j] += matrix1[i][k] * matrix2[k][j];
                    }
                }
            });

            return result;
        }
        public async static Task<int[][]> GenerateRandomMatrix(int size, int minVal, int maxVal)
        {
            int[][] matrix = new int[size][];

            Random random = new Random();

            for (int i = 0; i < size; i++)
            {
                matrix[i] = new int[size];

                for (int j = 0; j < size; j++)
                {
                    matrix[i][j] = random.Next(minVal, maxVal + 1);
                }
            }

            return matrix;
        }
    }
}

File: lab07/src/Methods.java
public class Methods {
    public static double[][] generateMatrix(int size){
        double[][] matrix = new double[size][size];
        for (int i = 0; i < size; i++) {
            double[] row = new double[size];
            for (int j = 0; j < size; j++) {
                row[j] = 1;
            }
            matrix[i] = row;
        }
        return matrix;
    }

    public static double[][] multiply(double[][] matrix1, double[][] matrix2){
        double[][] result = new double[matrix1.length][matrix2[0].length];
        for (int i = 0; i < matrix1.length; i++) {
            for (int j = 0; j < matrix1[0].length; j++) {
                double sum = 0;
                for (int k = 0; k < matrix1[0].length; k++) {
                    sum += matrix1[i][k] * matrix2[k][j];
                }
                result[i][j] = sum;
            }
        }
        return result;
    }

    public static void print(double[][] matrix){
        for(int i = 0 ;i < matrix.length; i++){
            for(int j = 0; j < matrix[i].length; j++){
                System.out.print(matrix[i][j] + " ");
            }
            System.out.println();
        }
    }
}